legend-engine-xt-relationalStore-executionPlan-connection/pom.xml

```diff
@@ -257,6 +263,13 @@
         </dependency>
         <!-- Authentication -->
 
+        <!-- Lombok for boilerplate code generation -->
```

Comment:
why this dependency?

Reply:
It is to omit boilerplate code.
Take a look how it is used:
https://github.com/roasher/legend-engine/blob/8f7a0d9de12ca87ef413025b5af26754fd55c69c/legend-engine-xt-relationalStore-executionPlan-connection/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/ds/specifications/keys/SpannerDataSourceSpecificationKey.java#L20
and compare to that:
https://github.com/roasher/legend-engine/blob/6869b7a93e5033244ab4b45a4f5ada39491173ed/legend-engine-xt-relationalStore-executionPlan-connection/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/ds/specifications/keys/BigQueryDataSourceSpecificationKey.java#L21

Constructors, equals, hashCode and toString() are generated in compile time (there is no lombok dep in production) makes simple pojos easy to maintain.

Reply:
ok, we should either use lombok in a consistent manner across the codebase, or not use it. Maybe we should not use it for now, and have a separate email thread to discuss using it across project?

Reply:
> ok, we should either use lombok in a consistent manner across the codebase, or not use it. Maybe we should not use it for now, and have a separate email thread to discuss using it across project?

I think we defenetily need a separete thread to discuss using it across project, but if it is passed I would suggest to leave it here in that way, shrinking future work and take benefits from it now, along task creation on using it across project. What to you think?